tests/unit/lms/services/canvas_api_test.py

```diff
@@ -465,18 +469,18 @@ def access_token_no_refresh(self, db_session, access_token_fields):
         return access_token
 
     @pytest.fixture
-    def api_client(self, api_client):
-        with mock.patch.object(api_client, "_get_refreshed_token"):
-            with mock.patch.object(api_client, "_validated_response"):
-                yield api_client
+    def base_client(self, base_client):
+        with mock.patch.object(base_client, "_get_refreshed_token"):
+            with mock.patch.object(base_client, "_validated_response"):
```

Comment:
I think if some of the tests for `FooClass` are having to patch some of `FooClass`'s own methods, that suggests that there's more than one separate class here. I wonder if `_get_refreshed_token` and `_validated_response` belong in separate collaborator classes, or just as separate standalone functions.

Maybe for a later PR